### PR TITLE
Fix y axes rendering and style for 3rd y axis

### PR DESCRIPTION
**Pull request by shefalijoshi**
### Describe your changes:
Iterate over all y axes for an overlay plot to display only the axes that have series associated with them
Add a class to show the 3rd y axes to the right of the canvas

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/plot/MctPlot.vue [hunk 1]
```diff
@@ -735,20 +760,24 @@ export default {
             }
         },
 
-        onTickWidthChange(width, fromDifferentObject) {
-            if (fromDifferentObject) {
+        onTickWidthChange(data, fromDifferentObject) {
```
Comment (ozyx) [inline src/plugins/plot/MctPlot.vue:763]: We could update the parameters to be `onTickWidthChange({ width, yAxisId }, fromDifferentObject)`

File: src/plugins/plot/axis/YAxis.vue [hunk 2]
```diff
@@ -113,18 +119,20 @@ export default {
...
-            if (isMainYAxis) {
-                if (this.visibleYAxes > 1) {
-                    style = `${width}; left: ${this.plotWidth - this.tickWidth + 40}px`;
+            const multipleTickOffset = this.multipleLeftAxes ? 20 : 0;
+            const border = this.multipleLeftAxes ? `border-right: 1px solid` : '';
+
+            if (this.position === 'left') {
+                const offset = (this.id - 1);
+                if (offset) {
+                    style = `${width}; ${border}; left: ${this.plotLeftTickWidth - this.tickWidth - offset * (multipleTickOffset + 5)}px`;
```
Comment (ozyx) [inline src/plugins/plot/axis/YAxis.vue:130]: Let's convert this to return a style object and clean up the magic numbers, see above comment on xAxisStyle

File: src/plugins/plot/axis/YAxis.vue [hunk 3]
```diff
@@ -113,18 +119,20 @@ export default {
...
                 } else {
-                    style = `${width}; left: ${this.plotWidth - this.tickWidth}px`;
+                    style = `${width}; left: ${this.plotLeftTickWidth - this.tickWidth + multipleTickOffset}px`;
                 }
-            } else {
-                style = `${width}; ${border}; left: ${this.plotWidth - this.tickWidth - 3}px`;
+            } else if (this.position === 'right') {
```
Comment (ozyx) [inline src/plugins/plot/axis/YAxis.vue:134]: can this just be an `else` if the position has a default value of 'left' ?

**Review by ozyx — CHANGES_REQUESTED** [review]
Looking good! Left a few suggestions mostly focused on readability and code cleanup. Tested it and the 3rd Axis is always appearing on the right of the plot, so that's awesome.

File: src/plugins/plot/MctPlot.vue [hunk 4]
```diff
@@ -461,15 +486,15 @@ export default {
...
         removeSeries(plotSeries, index) {
             const yAxisId = plotSeries.get('yAxisId');
-            this.setAxisVisibility(yAxisId, false);
+            this.updateAxisUsageCount(yAxisId, -1);
             this.seriesModels.splice(index, 1);
             this.stopListening(plotSeries);
         },
 
-        setAxisVisibility(yAxisId, visible) {
+        updateAxisUsageCount(yAxisId, updateCount) {
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:494]: I don't think that's necessary. It's a very simple method already.

File: src/plugins/plot/MctPlot.vue (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:763]: I tried that, but got confused while coding, so I changed it :)

File: src/plugins/plot/axis/YAxis.vue (hunk 3 above)
Comment (shefalijoshi) [reply src/plugins/plot/axis/YAxis.vue:134]: Done

File: src/plugins/plot/axis/YAxis.vue (hunk 2 above)
Comment (shefalijoshi) [reply src/plugins/plot/axis/YAxis.vue:130]: Done

File: src/plugins/plot/MctPlot.vue [hunk 5]
```diff
@@ -292,16 +296,24 @@ export default {
...
+            const hasRightAxis = this.yAxesIds.find(yAxis => yAxis.id > 2);
+            const leftOffset = this.multipleLeftAxes ? 40 : 20;
 
-            if (this.yAxesIds.length > 1) {
-                style = `left: ${this.plotWidth + 60}px`;
+            if (hasRightAxis) {
+                const rightTickWidth = this.yAxes[2].tickWidth;
+                style = `left: ${this.plotLeftTickWidth + leftOffset}px; right: ${rightTickWidth + 20}px`;
+            } else {
+                style = `left: ${this.plotLeftTickWidth + leftOffset}px`;
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:307]: Done

File: src/plugins/plot/MctPlot.vue [hunk 6]
```diff
@@ -292,16 +296,24 @@ export default {
...
         xAxisStyle() {
-            let style = `left: ${this.plotWidth + 20}px`;
+            let style;
+            const hasRightAxis = this.yAxesIds.find(yAxis => yAxis.id > 2);
+            const leftOffset = this.multipleLeftAxes ? 40 : 20;
 
-            if (this.yAxesIds.length > 1) {
-                style = `left: ${this.plotWidth + 60}px`;
+            if (hasRightAxis) {
+                const rightTickWidth = this.yAxes[2].tickWidth;
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:304]: Done

File: src/plugins/plot/MctPlot.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (ozyx) [reply src/plugins/plot/MctPlot.vue:494]: I guess just in terms of readability, when I see that function signature I would assume the function is updating the axisUsage count to that number we pass in, but it's actually incrementing or decrementing the current number based on what we pass in.

File: src/plugins/plot/axis/YAxis.vue [hunk 7]
```diff
@@ -119,20 +122,21 @@ export default {
...
+            };
+            const multipleAxesPadding = this.multipleLeftAxes ? AXIS_PADDING : 0;
+
+            if (this.position === 'right') {
+                style.left = `-${this.tickWidth + AXIS_PADDING}px`;
+            } else {
+                const thisIsTheSecondLeftAxis = (this.id - 1) > 0;
+                if (this.multipleLeftAxes && thisIsTheSecondLeftAxis) {
+                    style.left = `${ this.plotLeftTickWidth - this.tickWidth - multipleAxesPadding - AXIS_OFFSET }px`;
+                    style['border-right'] = `1px solid`;
```
Comment (ozyx) [inline src/plugins/plot/axis/YAxis.vue:136]: We can set these using their camelCase equivalents:

```js
style.borderRight = `1px solid`;
```

File: src/plugins/plot/MctPlot.vue [hunk 8]
```diff
@@ -296,15 +297,14 @@ export default {
     },
     computed: {
         xAxisStyle() {
-            let style;
-            const hasRightAxis = this.yAxesIds.find(yAxis => yAxis.id > 2);
-            const leftOffset = this.multipleLeftAxes ? 40 : 20;
+            const rightAxis = this.yAxesIds.find(yAxis => yAxis.id > 2);
+            const leftOffset = this.multipleLeftAxes ? 2 * AXES_PADDING : AXES_PADDING;
```
Comment (ozyx) [inline src/plugins/plot/MctPlot.vue:301]: Super readable, awesome!

**Review by ozyx — CHANGES_REQUESTED** [review]
Great stuff! I left just a couple comments but otherwise this is good to go.

File: src/plugins/plot/MctPlot.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue]: I changed the parameter name to updateCountBy

**Pull request merged**